qcodes/instrument_drivers/Keysight/keysightb1500/KeysightB1517A.py

```diff
@@ -369,6 +371,32 @@ def _set_post_sweep_voltage_condition(
         msg = MessageBuilder().wm(abort=self.sweep_auto_abort(), post=val)
         self.write(msg.message)
 
+    def _get_sweep_auto_abort(self):
+        msg = MessageBuilder().lrn_query(
+            type_id=constants.LRN.Type.STAIRCASE_SWEEP_MEASUREMENT_SETTINGS
+        )
+        response = self.ask(msg.message)
+        match = re.search(r'WM(?P<abort_function>.+?),'
+                          r'(?P<output_after_sweep>.+?)'
+                          r'(;|$)',
+                          response)
+
+        resp_dict = match.groupdict()
+        return int(resp_dict['abort_function'])
+
+    def _get_post_sweep_voltage_condition(self):
+        msg = MessageBuilder().lrn_query(
+            type_id=constants.LRN.Type.STAIRCASE_SWEEP_MEASUREMENT_SETTINGS
```

Comment:
shall we reduce duplocation by extracting the common code into a method called, say, _get_STAIRCASE_SWEEP_MEASUREMENT_SETTINGS ? same for CV_DC_BIAS_SWEEP_MEASUREMENT_SETTINGS ?

Reply:
done